client/my-sites/domains/domain-management/settings/cards/connected-domain-details.tsx

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable wpcalypso/jsx-classname-namespace */
+
+import { Button } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import { useLocalizedMoment } from 'calypso/components/localized-moment';
+import { getManagePurchaseUrlFor } from 'calypso/my-sites/purchases/paths';
+import type { DetailsCardProps } from './types';
+
+import './style.scss';
+
+const ConnectedDomainDetails = ( {
+	domain,
+	isLoadingPurchase,
+	selectedSite,
+}: DetailsCardProps ): JSX.Element => {
+	const moment = useLocalizedMoment();
+	const translate = useTranslate();
+
+	const renderPlanDetailsButton = () => {
+		return (
+			<Button
+				href={ getManagePurchaseUrlFor( selectedSite.slug, domain.bundledPlanSubscriptionId ) }
```

Comment:
Here you can optionally use the non null assertion (I know, linter will still complain, but a little less 😆):

`href={ getManagePurchaseUrlFor( selectedSite.slug, domain.bundledPlanSubscriptionId! ) }`

Reply:
Just out of curiosity, how do you know that `domain.bundledPlanSubscriptionId` will not be `null` or `undefined`? Because if you do know that, you can just make it explicit by narrowing the type before this line instead of using the (slightly unusual) assertion operator.

For example:

```ts
	const renderPlanDetailsButton = () => {
		if ( ! domain.bundledPlanSubscriptionId ) {
			throw new Error( 'Cannot render plan details for domainw ithout bundled plan' );
		}
		return (
			<Button
				href={ getManagePurchaseUrlFor( selectedSite.slug, domain.bundledPlanSubscriptionId ) }
				disabled={ isLoadingPurchase }
			>
				{ translate( 'View plan settings' ) }
			</Button>
		);
	};
```

Reply:
This is a good suggestion, thanks! Gotta improve my Typescript knowledge 🙂

Reply:
Let me know if I can help! Getting types right can definitely be tricky sometimes. Feel free to ping me on Slack if you have more detailed questions than I can answer here. 😁

Reply:
Thanks @sirbrillig! Actually I saw your first comment after replying to Igor's comment, hadn't refreshed the page when I posted my reply 😅 but yeah, this component is only shown for mapped domains, so I believe `domain.bundledPlanSubscriptionId` should probably be defined. But I think adding an early return if it's not is a good idea 👍